src/lamp_py/ingestion_tm/jobs/parition_table.py

```diff
@@ -0,0 +1,189 @@
+        """write version file to s3 for partition dataset"""
+        with tempfile.TemporaryDirectory() as temp_dir:
+            local_version = os.path.join(temp_dir, self.version_key)
+
+            with open(local_version, "w", encoding="utf8") as f:
+                f.write(self.lamp_version)
+
+            upload_file(
+                file_name=local_version,
+                object_path=self.s3_version_path,
+                extra_args={"Metadata": {self.version_key: self.lamp_version}},
+            )
+
+    def dates_from_tm(self, tm_db: MSSQLManager) -> List[int]:
+        """
+        retrieve dates to process from Transit Master database
+
+        returns sorted list of CALENDAR_ID dates
+
+        :return [120240101, 120240102]
+        """
+        tm_dates_query = sa.text(
+            "SELECT DISTINCT CALENDAR_ID FROM TMDailyLog.dbo.STOP_CROSSING;"
+        )
+        tm_dates = tm_db.select_as_dataframe(tm_dates_query)
+
+        return sorted(tm_dates["CALENDAR_ID"].astype(int).to_list())
+
+    def dates_from_s3(self) -> List[int]:
+        """
+        retrive list of already exported dates from S3 prefix
+
+        returns sorted list of CALENDAR_ID extracted from paths
+
+        :return [120240101, 120240102]
+        """
+        s3_files = file_list_from_s3(self.export_bucket, self.s3_export_prefix)
+
+        def date_match(s: str) -> Optional[int]:
+            match = re.search(r"\d{9}", s)
+            if match:
+                return int(match.group())
+            return None
+
+        return sorted([s for s in map(date_match, s3_files) if s])
+
+    def dates_to_export(self, tm_db: MSSQLManager) -> List[int]:
+        """
+        compare S3 exported file dates to available dates in Transit Master
+        export any dates from Transit Master that are not found in S3 as well as
+        the last two dates found in Transit Master
+
+        if expected lamp_version does not match, return all available Transit Master dates
+
+        :return [120240101, 120240102]
+        """
+        s3_version = None
+        tm_dates = self.dates_from_tm(tm_db)
+
+        version_file = file_list_from_s3(
+            self.export_bucket,
+            os.path.join(self.s3_export_prefix, self.version_key),
+        )
+        if len(version_file) == 1:
+            s3_version = object_metadata(self.s3_version_path).get(
+                self.version_key
+            )
+
+        if s3_version != self.lamp_version:
+            return tm_dates
+
+        s3_dates = self.dates_from_s3()
+
+        export_dates = set(tm_dates).difference(set(s3_dates))
+        export_dates.update(tm_dates[-2:])
+
+        return sorted(export_dates)
+
+    def run_export(self, tm_db: MSSQLManager) -> None:
+        table_columns = ",".join([col.name for col in self.export_schema])
+
+        logger = ProcessLogger("tm_stop_crossing_export")
+        logger.log_start()
+        try:
+            for date in self.dates_to_export(tm_db):
```

Comment:
All of the semi-complicated logic is in `dates_to_export`. for each date the operation is essentially the same as for the whole table export, build the query, create a local parquet file and upload to S3.